lib/experimental/new_embed.dart

```diff
@@ -191,6 +198,49 @@ class EditorTabView extends TabView {
       Timer(const Duration(seconds: 0), _editor.resize);
     }
   }
+
+  /// Perform static analysis of the source code.
+  void _performAnalysis(_) async {
```

Comment:
I wonder if there's a more elegant way to debounce these calls that creating a timer manually. LGTM for now, though.

Reply:
I can't think of one at this point. Happy to take pointers to other approaches, but this is a fairly common approach, especially for people who have come up through embedded programming.

Reply:
It is worth noting that the playground itself is debouncing after the network round trip, which means that the servers are doing a lot more work. We should consider implementing something akin to this debouncing technique in the playground as well.

Reply:
And here's an interesting abstraction: https://davidwalsh.name/javascript-debounce-function

Reply:
It looks like `_performAnalysis` is called after every doc change, then debounces itself, then makes a network request.

I'm surprised we didn't have something similar for the other dartpad - I'll take a look.

Reply:
It's not as straightforward, but if you looks for `_createReconciler()`, we were debouncing on edits to the code at 1,250ms.

Reply:
You can also see `DelayedTimer` - https://github.com/flutter/devtools/blob/master/packages/devtools/lib/src/utils.dart#L154 - from the devtools codebase. It will debounce if new events come in during a min. threshold, but will emit an event if a max delay is hit.

Reply:
Ahh, thanks for finding that for me Devon =)